compiler/noirc_frontend/src/hir/comptime/value.rs

```diff
@@ -397,6 +397,29 @@ impl Value {
             value => Err(InterpreterError::CannotInlineMacro { value, location }),
         }
     }
+
+    /// Creates a value that holds an `Option``.
+    /// `option_type` must be a Type referencing the `Option` type.
+    pub(crate) fn option(option_type: Type, value: Option<Value>) -> Value {
+        let Type::Struct(shared_option_type, mut generics) = option_type.clone() else {
+            panic!("Expected type to be a struct");
+        };
+
+        let shared_option_type = shared_option_type.borrow();
+        assert_eq!(shared_option_type.name.0.contents, "Option");
+
+        let t = generics.pop().expect("Expected Option to have a T generic type");
+
+        let (is_some, value) = match value {
+            Some(value) => (Value::Bool(true), value),
+            None => (Value::Bool(false), Value::Zeroed(t)),
```

Comment:
Please avoid using `Value::Zeroed`. Prefer using the `zeroed` builtin in interpreter/builtin.rs since it will return an actual zero value more often. Using `Value::Zeroed` will lead to very weird errors if users ever actually use these values (e.g. creating an array of zeroes).